File: TenMinsJava/day55.java
// Check if string contains 8 digits (even if there's dash)

package TenMinsJava;

import java.util.Scanner;

public class day55 {
    public static int digit = 0;
    public static int maxL = 8;
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        System.err.println("Enter 8 digit number (- is ignored)");
        String s = scan.nextLine();
        for (Character c : s.toCharArray()) {
            if (c == '-') { maxL++; }
            if (Character.isDigit(c)) { digit++; }
        }
        if (s.length() == maxL && digit == 8) {
            System.err.println("nice");
        }
        else {
            System.err.println("nah");
        }
    }
}

File: TenMinsPy/day55.py
# Check if string contains 8 digits (even if there's dash)

import re

class day55:
    def __init__(self):
        s = input("Enter 8 digit number (- is ignored)\n")
        maxL = 8
        digit = 0
        for c in s:
            if c.isdigit(): 
                digit += 1
        for c in s:
            if (c == '-' ):
                maxL += 1
        if len(s) == maxL:
            print('nice')
        else:
            print('is not 8 digit or not a digit')

day55()